modules/formulize/class/screen.php

```diff
@@ -1,421 +1,422 @@
+###############################################################################
+##                    XOOPS - PHP Content Management System                  ##
+##                       Copyright (c) 2000 XOOPS.org                        ##
+##                          <http://www.xoops.org/>                          ##
+###############################################################################
+##  This program is free software; you can redistribute it and/or modify     ##
+##  it under the terms of the GNU General Public License as published by     ##
+##  the Free Software Foundation; either version 2 of the License, or        ##
+##  (at your option) any later version.                                      ##
+##                                                                           ##
+##  You may not change or alter any portion of this comment or credits       ##
+##  of supporting developers from this source code or any supporting         ##
+##  source code which is considered copyrighted (c) material of the          ##
+##  original comment or credit authors.                                      ##
+##                                                                           ##
+##  This program is distributed in the hope that it will be useful,          ##
+##  but WITHOUT ANY WARRANTY; without even the implied warranty of           ##
+##  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            ##
+##  GNU General Public License for more details.                             ##
+##                                                                           ##
+##  You should have received a copy of the GNU General Public License        ##
+##  along with this program; if not, write to the Free Software              ##
+##  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA ##
+###############################################################################
+##  Author of this file: Freeform Solutions                                  ##
+##  Project: Formulize                                                       ##
+###############################################################################
+
+if (!defined("XOOPS_ROOT_PATH")) {
+    die("XOOPS root path not defined");
+}
+
+require_once XOOPS_ROOT_PATH.'/kernel/object.php';
+class formulizeScreen extends xoopsObject {
+
+	function __construct() {
+        parent::__construct();
+		$this->initVar('sid', XOBJ_DTYPE_INT, '', true);
+		$this->initVar('title', XOBJ_DTYPE_TXTBOX, '', true, 255);
+		$this->initVar('fid', XOBJ_DTYPE_INT, '', true);
+		$this->initVar('frid', XOBJ_DTYPE_INT, '', true);
+		$this->initVar('type', XOBJ_DTYPE_TXTBOX, '', true, 100);
+		$this->initVar('useToken', XOBJ_DTYPE_INT);
+    $this->initVar('anonNeedsPasscode', XOBJ_DTYPE_INT);
+    $this->initVar('theme', XOBJ_DTYPE_TXTBOX, '', true, 100);
+		$this->initVar('rewriteruleAddress', XOBJ_DTYPE_TXTBOX, '', false, 255);
```

Comment:
There doesn't seem to be any validation of input on this field. I was able to enter characters that are not valid in URLs including emoji. Additionally spaces were allowed and did not correctly work with %20 character encoding. I think we need very strict validation on this feeld as it's a major footgun for administrators.

Reply:
So true!

Reply:
Done now, by converting spaces to _ and stripping out non alphanumerics, non underscores, and non hyphens.